include/SFML/Graphics/Glsl.inl

```diff
@@ -0,0 +1,203 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2015 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////
+/// \brief Helper function to copy sf::Transform to sf::Glsl::Mat3
+///
+////////////////////////////////////////////////////////////
+void initializeMatrix(Matrix<3, 3>& matrix, const Transform& transform);
+
+
+////////////////////////////////////////////////////////////
+/// \brief Helper function to copy sf::Transform to sf::Glsl::Mat4
+///
+////////////////////////////////////////////////////////////
+void initializeMatrix(Matrix<4, 4>& matrix, const Transform& transform);
+
+
+////////////////////////////////////////////////////////////
+/// \brief Copy array-based matrix with given number of elements
+///
+/// Indirection to std::copy() to avoid inclusion of
+/// <algorithm> and MSVC's annoying 4996 warning in header
+///
+////////////////////////////////////////////////////////////
+void copyMatrix(const float* source, std::size_t elements, float* dest);
```

Comment:
Is there a reason for this parameter ordering? I find following the same ordering as `std::memcpy` to be more "intuitive" and probably easier for people to remember.